include/ccf/byte_vector.h

```diff
@@ -40,7 +40,8 @@ struct formatter<ccf::ByteVector>
   template <typename FormatContext>
   auto format(const ccf::ByteVector& e, FormatContext& ctx) const
   {
-    if (std::find(e.begin(), e.end(), '\0') != e.end())
+    auto non_printable = [](uint8_t b) { return b < 0x20 || b > 0x7e; };
+    if (std::find_if(e.begin(), e.end(), non_printable) != e.end())
```

Comment:
I explicitly didn't want to use `isprint`, since that is defined as "locale dependent" (whatever that means on OE?), hence the custom implementation.

I agree using `std::all` would still be clearer than `find_if` so I'll switch to that.